### DISCUSSION TITLE
Validation to ensure no "extra" fields are present?

### DISCUSSION THREAD

**Discussion by jonmmyers** [go-playground/validator#1038] [Q&A]
For my use case, I need to validate that there are _no additional fields_ other than those laid out in the schema. In this case, my data is of type `map[string]interface{}` and I'm using schemas defined in .json files that get read into a `map[string]interface{}` before calling the `ValidateMap` method, eg:

```json
{
  "amount": "required,numeric",
  "type": "required,oneof=debit credit",
  "currency": {
    "code": "required,alpha,uppercase,len=3",
    "name": "required,uppercase"
    }
}
```

So I would want this data to pass the validation:
```go
map[string]interface{}{
	"amount": 12,
        "type":   "debit",
	"currency": map[string]interface{}{
		"code": "USD",
		"name": "UNITED STATES DOLLAR",
	},
}

```

And this rejected:
```go
map[string]interface{}{
	"amount": 12,
        "type":   "debit",
	"currency": map[string]interface{}{
		"code": "USD",
		"name": "UNITED STATES DOLLAR",
                "extraFieldName": "EXTRA DATA",
	},
}

```

Is this possible using the validator package? Thanks!

**Comment by josepmdc** [comment]
Hi! I'm trying to do something similar, did you find a solution?
